ruby_programming/computer_science/project_data_structures_algorithms.md

```diff
@@ -24,7 +24,7 @@ You'll build a simple binary search tree in this assignment. In this lesson, our
 
   6. Write `#inorder`, `#preorder`, and `#postorder` methods which accept a block. Each method should traverse the tree in their respective depth-first order and yield each node to the provided block. As a bonus, make the method return an array of values if no block is given.
 
-  7. Write a `#depth` method which accepts a node and returns the depth(number of levels) beneath the node.
+  7. Write a `#height` method which accepts a node and returns the height of the node. Height is defined as the number of edges in longest path from a given node to a leaf node.
```

Comment:
Grammar-wise, can we change the second part of the first phrase to `and returns its height`? We're already referring to `node` in the first half, so this reduces redundancy.

Otherwise LGTM. Thanks @Tenacious-Qi

Reply:
I personally find this more confusing, I always use the root node as a reference point and almost think the tree spreading downwards (depth). In which case the original definition I think was just missing the word "root): 

```
Write a `#depth` method which accepts a node and returns the depth(number of levels) beneath the root node.
```

Per your linked stacked overflow, depth and height are just the inverse of each other depending on whether you're talking about root node to a leaf or a leaf node to root so neither is more correct. Maybe we should mention both since I would find thinking of height as more confusing than depth?

Reply:
Maybe we should have a method for both depth and height since they're just inverses depending on how you view tree.

Reply:
@leosoaivan Thanks that is better that way, new commit submitted. I had weird things happen when I tried to pull upstream master, so there is an extra commit in there with no changes.

Reply:
@Drew-Goddyn good idea. Should that be a separate PR? I'm not really sure. My main concern was that step 8 mentions depth and then #balanced? in step 9 mentions height. That inconsistency is confusing if you don't yet know depth and height are inverses. But, as you said, having both methods required could help clarify the difference and then students could decide which they wanted to use in their code.

Reply:
@Tenacious-Qi sorry for delay. It can be in a followup pr if you prefer since this isn't _technically_ incorrect